app/background/wallet/create-register-all.js

```diff
@@ -0,0 +1,65 @@
+import walletClient from "../../utils/walletClient";
+
+const MTX = require('hsd/lib/primitives/mtx');
+const {states} = require('hsd/lib/covenants/namestate');
+const rules = require('hsd/lib/covenants/rules');
+const {types} = rules;
+const Output = require('hsd/lib/primitives/output');
+const EMPTY = Buffer.alloc(0);
+
+export default async function createRegisterAll(wallet) {
+  const height = wallet.wdb.height + 1;
+  const network = wallet.network;
+  const names = await wallet.txdb.getNames();
+  const mtx = new MTX();
+
+  for (let i = 0; i < names.length; i++) {
+    const ns = names[i];
+    const {owner} = ns;
+
+    const coin = await wallet.getCoin(owner.hash, owner.index);
+
+    if (coin) {
+      if (!coin)
+        continue;
+
+      if (ns.isExpired(height, network))
+        continue;
+
+      // Is local?
+      if (coin.height < ns.height)
+        continue;
+
+      if (!coin.covenant.isReveal() && !coin.covenant.isClaim())
+        continue;
+
+      if (coin.covenant.isClaim()) {
+        if (height < coin.height + network.coinbaseMaturity)
+          continue;
+      }
+
+      const state = ns.state(height, network);
+
+      if (state !== states.CLOSED)
+        continue;
+
+      const output = new Output();
+      output.address = coin.address;
+      output.value = ns.value;
+
+      output.covenant.type = types.REGISTER;
+      output.covenant.pushHash(ns.nameHash);
+      output.covenant.pushU32(ns.height);
+      output.covenant.push(EMPTY);
```

Comment:
Hey nice, this is really helpful! Not sure if this is the right place, but is it possible to set records here? Maybe with (non-persisting) user input. Personally, I set it to Namebase DNS so it just works after transferring to Namebase.

Reply:
Yep, this would be the right place to set records here. For UI/UX i would use the multi-select modal from bulk transfer #315